pika/adapters/select_connection.py

```diff
@@ -264,21 +264,38 @@ def get_next_deadline(self):
         timeout = min((timeout, SelectPoller.POLL_TIMEOUT))
         return timeout * SelectPoller.POLL_TIMEOUT_MULT
 
+    @staticmethod
+    def is_ready_to_fire(to_run):
+        """For patching by unittest"""
+        return True
+
     def process_timeouts(self):
         """Process the self._timeouts event stack"""
 
         now = time.time()
-        to_run = [timer for timer in self._timeouts.values()
-                  if timer['deadline'] <= now]
-
         # Run the timeouts in order of deadlines. Although this shouldn't
         # be strictly necessary it preserves old behaviour when timeouts
         # were only run periodically.
-        for t in sorted(to_run, key=itemgetter('deadline')):
```

Comment:
@vitaly-krugl Thank you very much for your replies. I'll follow your advice of selectively disabling the warningings thought rather than using setattr() instead of regular assignment. As for my question on pylint complaining about assertEqual, it was my misunderstanding. My apology.